src/Interpreters/InterpreterRenameQuery.cpp

```diff
@@ -76,8 +76,20 @@ BlockIO InterpreterRenameQuery::executeToTables(const ASTRenameQuery & rename, c
 
     for (const auto & elem : descriptions)
     {
-        if (!rename.exchange)
+        bool exchange_tables = rename.exchange;
+        if (exchange_tables)
+        {
+            bool allow_rename_instead_of_exchange = descriptions.size() == 1 && rename.rename_if_cannot_exchange;
```

Comment:
Why do we have the same interpreter for the RENAME query and for EXCHANGE... This boolean flags are very difficult to understand :(